### PR TITLE
Add keyPairFromSeed

### PR DESCRIPTION
**Pull request by mrcnski**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ export class SkynetClient {
 
   // SkyDB
   db = {
+    keyPairFromSeed: keyPairFromSeed.bind(this),
```
Comment (peterjan) [inline src/client.ts]: I would not define this on the db.

Comment (interstella-5555) [reply src/client.ts]: agreed, it's also something registry will make use of

File: src/skydb.ts [hunk 2]
```diff
@@ -66,7 +66,13 @@ export async function setJSON(
     try {
       entry = await this.registry.getEntry(publicKey, dataKey, opts);
 
-      // verify here
+      revision = entry.entry.revision + 1;
+    } catch (err) {
+      revision = 0;
+    }
+
+    if (entry) {
```
Comment (interstella-5555) [inline src/skydb.ts]: flatten those two ifs into one if

File: src/utils.ts [hunk 3]
```diff
@@ -55,6 +57,15 @@ export function getRootDirectory(file: File): string {
   return path.normalize(dir).slice(root.length).split(path.sep)[0];
 }
 
+/**
+ * Generates a public and private key from a provided, secure seed.
+ * @param seed - A secure seed.
+ */
+export function keyPairFromSeed(seed: string): { publicKey: PublicKey; privateKey: SecretKey } {
+  seed = pkcs5.pbkdf2(seed, "", 1000, 32, md.sha256.create());
```
Comment (interstella-5555) [inline src/utils.ts:66]: Can we add a comment on why is this needed here ? @peterjanbrone maybe you could collaborate on that

File: src/integration.test.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
-import { SkynetClient } from "./client";
-import { pki } from "node-forge";
+import { keyPairFromSeed, SkynetClient } from "./index";
 
-const { publicKey, privateKey } = pki.ed25519.generateKeyPair();
+const { publicKey, privateKey } = keyPairFromSeed("test");
```
Comment (interstella-5555) [inline src/integration.test.ts]: this should have a randomly generated seed

File: src/utils.ts [hunk 5]
```diff
@@ -55,6 +57,15 @@ export function getRootDirectory(file: File): string {
   return path.normalize(dir).slice(root.length).split(path.sep)[0];
 }
 
+/**
+ * Generates a public and private key from a provided, secure seed.
+ * @param seed - A secure seed.
+ */
+export function keyPairFromSeed(seed: string): { publicKey: PublicKey; privateKey: SecretKey } {
```
Comment (interstella-5555) [inline src/utils.ts:64]: I would prefix that function name with a verb like `generate` or `create`

**Review by interstella-5555 — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 5 above)
Comment (mrcnski) [reply src/utils.ts:64]: I think it's fine as-is.

File: src/integration.test.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
-import { keyPairFromSeed, SkynetClient } from "./index";
+import { SkynetClient } from "./client";
+import { pki } from "node-forge";
 
-const { publicKey, privateKey } = keyPairFromSeed("test");
+const { publicKey, privateKey } = pki.ed25519.generateKeyPair();
```
Comment (interstella-5555) [inline src/integration.test.ts]: this is not what I meant, you should still use `keyPairFromSeed` but the seed you pass should be random instead of `"test"`

**Review by interstella-5555 — CHANGES_REQUESTED** [review]

**Review by interstella-5555 — APPROVED** [review]
